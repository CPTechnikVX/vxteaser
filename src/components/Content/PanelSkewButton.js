import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';
import ClickEvent from '../../utils/ClickEvent';

/**
 * Skewed button for skewed panel
 */
export default class PanelSkewButton extends React.PureComponent {
	render() {
		const {children, config, link, modifier, windowWidth} = this.props;
		const classList                                       = [];
		const styleObj                                        = {};
		let onClickFn;

		classList.push(Constants.ClassName.PanelSkewButton);

		if (link) {
			onClickFn = (e) => {
				e.preventDefault();
				e.stopPropagation();

				this.props.onClickFn(new ClickEvent(e, this, {
					link: this.props.link,
				}));
			};
		}

		if (modifier) {
			classList.push(modifier);
		}

		if (windowWidth < config.fixedHeights[1]['greaterThan']) {
			styleObj.top = `-${config.fixedHeights[1]['height'] / 2.71}px`;
		}

		return <a className={classnames(classList)} onClick={onClickFn} style={styleObj}>{children}</a>;
	}
}

PanelSkewButton.propTypes = {
	/** @ignore */
	children:    PropTypes.node,
	/** @ignore */
	config:      PropTypes.object,
	/**
	 * Textual color definition
	 */
	color:       PropTypes.oneOf(['primary']),
	link:        PropTypes.string,
	modifier:    PropTypes.string,
	/** @ignore */
	onClickFn:   PropTypes.func,
	/** @ignore */
	windowWidth: PropTypes.number,
};
