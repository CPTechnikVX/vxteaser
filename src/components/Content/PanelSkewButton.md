```html
<PanelSkewButton link="https://www.example.com/">Skewed Button</PanelSkewButton>
```
