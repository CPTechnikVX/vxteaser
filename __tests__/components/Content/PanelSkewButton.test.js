import React           from 'react';
import {shallow}       from 'enzyme';
import PanelSkewButton from '../../../src/components/Content/PanelSkewButton';

describe('render panel skew button', () => {
	const config = {
		fixedHeights: [
			{},
			{
				greaterThan: 1200,
				height:      260,
			},
		],
	};

	test('desktop', () => {
		const wrapper = shallow(
			<PanelSkewButton link="http://www.example.com" modifier="first" config={config} windowWidth={1920}>Content</PanelSkewButton>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('tablet', () => {
		const wrapper = shallow(
			<PanelSkewButton link="http://www.example.com" modifier="first" config={config} windowWidth={768}>Content</PanelSkewButton>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
