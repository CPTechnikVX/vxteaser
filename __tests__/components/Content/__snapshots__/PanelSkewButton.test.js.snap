// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render panel skew button desktop 1`] = `
<a
  className="vxteaser-skew-button first"
  onClick={[Function]}
  style={Object {}}
>
  Content
</a>
`;

exports[`render panel skew button tablet 1`] = `
<a
  className="vxteaser-skew-button first"
  onClick={[Function]}
  style={
    Object {
      "top": "-95.9409594095941px",
    }
  }
>
  Content
</a>
`;
